src/NHapi.Base/Parser/XMLParser.cs

```diff
@@ -25,796 +25,828 @@
+        /// </summary>
+        protected internal virtual string RemoveWhitespace(string s)
+        {
+            s = s.Replace('\r', ' ');
+            s = s.Replace('\n', ' ');
+            s = s.Replace('\t', ' ');
+
+            var repeatedSpacesExist = true;
+            while (repeatedSpacesExist)
+            {
+                var loc = s.IndexOf("  ");
+                if (loc < 0)
+                {
+                    repeatedSpacesExist = false;
+                }
+                else
+                {
+                    var buf = new StringBuilder();
+                    buf.Append(s.Substring(0, loc - 0));
+                    buf.Append(" ");
+                    buf.Append(s.Substring(loc + 2));
+                    s = buf.ToString();
+                }
+            }
+
+            return s.Trim();
+        }
+
+        /// <summary>
+        /// Parses a message string and returns the corresponding Message
+        /// object. This method checks that the given message string is XML encoded, creates an
+        /// XML Document object (using Xerces) from the given String, and calls the abstract
+        /// method <see cref="ParseDocument(XmlDocument, string)"/>.
+        /// </summary>
+        protected internal override IMessage DoParse(string message, string version)
+        {
+            IMessage m = null;
+
+            // parse message string into a DOM document
+            try
+            {
+                var doc = new XmlDocument();
+                doc.Load(new StringReader(message));
+
+                // rlb: Don't think we need to lock this...
+                // lock (this)
+                // {
+
+                // //UPDATE_NOT: Replaced the following
+                //    //parser.parse(new XmlSourceSupport(new System.IO.StringReader(message)));
+                //    //doc = parser.getDocument();
+                // }
+                m = ParseDocument(doc, version);
+            }
+            catch (XmlException e)
+            {
+                throw new HL7Exception("XmlException parsing XML", ErrorCode.APPLICATION_INTERNAL_ERROR, e);
+            }
+            catch (IOException e)
+            {
+                throw new HL7Exception("IOException parsing XML", ErrorCode.APPLICATION_INTERNAL_ERROR, e);
+            }
+
+            return m;
+        }
+
+        /// <summary> Formats a Message object into an HL7 message string using the given
+        /// encoding.
+        /// </summary>
+        /// <throws>  HL7Exception if the data fields in the message do not permit encoding. </throws>
+        /// <summary>      (e.g. required fields are null).
+        /// </summary>
+        /// <throws>  EncodingNotSupportedException if the requested encoding is not. </throws>
+        /// <summary>      supported by this parser.
+        /// </summary>
+        protected internal override string DoEncode(IMessage source, string encoding)
+        {
+            if (!encoding.Equals("XML"))

```

Comment:
there is a method for checking the encoding is supported.

```csharp
if (!SupportsEncoding("XML"))
{
    throw new EncodingNotSupportedException("XMLParser supports only XML encoding");
}

return Encode(source);
```

Reply:
done